mass2/core/channels.py

```diff
@@ -653,3 +654,57 @@ def from_df(
                 npulses=len(df),
             )
         return Channels(channels, description)
+
+    def save_analysis(self, path: Path | str, overwrite: bool = False) -> None:
+        """Save an analysis-in-progress completely to a directory
+
+        Parameters
+        ----------
+        path : Path | str
+            Directory to save work in. If it doesn't exist, its parent should.
+        overwrite : bool, optional
+            If `path` exists, whether to overwrite it, by default False
+        """
+        if isinstance(path, str):
+            path = pathlib.Path(path)
+        if os.path.exists(path):
+            if os.path.exists(path / "data_all.pkl") and not overwrite:
+                raise ValueError(f"Set save_analysis(...overwrite=True) to overwrite the existing {path=}")
+            if not os.path.isdir(path):
+                raise ValueError(f"save_analysis(path={path}) requires that `path` be a directory, if it exists")
+        else:
+            os.mkdir(path)
+            assert os.path.exists(path) and os.path.isdir(path)
+
+        channels = {}
+        for ch_num, ch in self.channels.items():
+            fname = path / f"data_chan{ch_num:04d}.parquet"
+            df = ch.df.drop("pulse", "timestamp", "subframecount")
+            df.write_parquet(fname)
+            steps = ch.steps.trim_dead_ends(None, drop_debug=True)
+            channels[ch_num] = dataclasses.replace(ch, df=pl.DataFrame(), df_history=[], noise=None, steps=steps)
```

Comment:
Should we save the noise? I guess its mostly useless since all it can do is use its `DataFrame` to calculate a spectrum. Any spectra we've calculated would live in `steps` as long as we don't drop debug info.

Reply:
Honestly, I'm inclined to remove the noise from a `Channel` in every situation, not just in the save-game case. I'm okay with keeping a noise analysis result, but I don't really like having a (pointer to a) complete noise channel. We can discuss offline--it's beyond the scope of this PR.